### PR TITLE
Watch other resources (#226)

### LINKED ISSUE #226: Additional resources in watch mode
**Enhancement** 
Now we have the ability to leverage node modules during build time render, we can build tooling that dynamically read resources that then get processed and included in the bundle. Because these resources are not directly/statically imported webpack does not currently watch them. This is a bit cumbersome if say working on a blog and updating some markdown, i'd like to be able to see my changes with a watch.

If we leverage https://github.com/pigcan/extra-watch-webpack-plugin, we should be able to watch everything other than node_modules in a dojo project.

### PR DESCRIPTION
**Pull request by maier49**
Use the extra-watch-webpack-plugin to watch all files not in
dot folders, output, or node_modules.

**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Resolves #226

### CODE REVIEW

File: src/base.config.ts [hunk 1]
```diff
@@ -401,6 +403,10 @@ export default function webpackConfigFactory(args: any): webpack.Configuration {
 							has: 'web-animations'
 						}
 					]
+				}),
+			watch &&
+				new ExtraWatchWebpackPlugin({
+					files: ['!(output|.*)/**']
```
Comment (matt-gadd) [inline src/base.config.ts]: does this ignore node_modules by default?

File: package.json [hunk 2]
```diff
@@ -114,6 +114,7 @@
     "eventsource-polyfill": "0.9.6",
     "express": "4.16.2",
     "express-static-gzip": "1.1.3",
+    "extra-watch-webpack-plugin": "^1.0.3",
```
Comment (matt-gadd) [inline package.json]: can we pin this?

File: src/base.config.ts [hunk 3]
```diff
@@ -417,7 +423,7 @@ export default function webpackConfigFactory(args: any): webpack.Configuration {
 					}
 				},
 				{
-					test: /\.(css|js)$/,
+					test: /\.(css|js)/,
```
Comment (matt-gadd) [inline src/base.config.ts]: was this meant to change?

File: src/base.config.ts (hunk 1 above)
Comment (maier49) [reply src/base.config.ts]: That was the behavior I observed in testing, but it's not explicitly documented as such so I've added it here.

**Pull request merged**